transaction-manifest/src/generator.rs

```diff
@@ -282,17 +282,27 @@ pub fn generate_instruction(
             args,
         } => {
             let args = generate_args(args, resolver)?;
+            let mut fields = Vec::new();
             for arg in &args {
                 let validated_arg = ScryptoValue::from_slice(arg).unwrap();
                 id_validator
                     .move_resources(&validated_arg)
                     .map_err(GeneratorError::IdValidatorError)?;
+                fields.push(validated_arg.dom);
             }
+
+            let variant = ::sbor::Value::Enum {
+                name: generate_string(function)?,
+                fields
+            };
+            let mut bytes = Vec::new();
+            let mut enc = ::sbor::Encoder::with_type(&mut bytes);
+            ::sbor::encode_any(None, &variant, &mut enc);
```

Comment:
I'm now convinced we should have a function for this :)

Reply:
ha yes, okay